### ISSUE TITLE
Ensure test titles are unique in each test file

### ISSUE THREAD

**Issue by krlmlr** [r-lib/testthat#1354]
Often, duplicate test titles are the result of a copy-paste operation, and the title should really be different. Duplicate test titles also break snapshots; this is already taken into account by snapshot tests, an early warning reduces confusion.

**Comment by hadley** [comment]
I think this would have to be done at the reporter level? You'd use `start_test()` to build up a list of test names, warn if you'd seen the name before, and then reset the list in `end_file()`. But that would tie it to a specific reporter, which seems inelegant. Maybe this is best thought of as linting and done elsewhere?

**Comment by krlmlr** [comment]
Should we add a "linting reporter" that is always included? That reporter could also warn e.g. if `test_that()` or `expect_snapshot()` are used without braces.

If we went for the linting approach, we could still use an "enumerating" reporter that enumerates all tests and their locations without actually running the code. (Last time I checked, implementing a custom reporter in a different package requires copy-pasting code from testthat, in particular related to the "expectation" class. Should that be easier?)
